### PR TITLE
Task-47492: Badges by domain is not displayed

### PR DESCRIPTION
**Pull request by mzorai**
Prior this change,gamification points for normal activity in space are not earned, this is due to NPE when checking activity type

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java [hunk 1]
```diff
@@ -283,7 +283,7 @@ public Response postActivity(
       activity.setTitle(model.getTitle());
     }
     activity.setBody(model.getBody());
-    activity.setType(model.getType());
+    activity.setType(model.getType() != null ? model.getType() : "");
```
Comment (boubaker) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java:286]: This seems to be a **workaround**, the Service & Storage layers has to handle this attribute when it's null. Same for gamification which should be improved to handle when type is null.

**Review by boubaker — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java (hunk 1 above)
Comment (ahamdi) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java:286]: @mzorai @AzmiTouil  Could you take care of the comment of @boubaker  ? The fix should be done on Gamification if the NPE is thrown there.

Comment (AzmiTouil) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java:286]: https://github.com/Meeds-io/gamification/pull/165 cc @ahamdi

Comment (AzmiTouil) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java:286]: https://github.com/Meeds-io/social/pull/1107

**Pull request closed without merging**